pkg/runctl/run_manager.go

```diff
@@ -97,60 +96,85 @@ func (c *runManager) prepareRunNamespace(pipelineRun k8s.PipelineRun) error {
 	defer cleanupOnError()
 
 	//Copy secrets to Run Namespace
+	targetClient := c.factory.CoreV1().Secrets(runNamespace)
+	secretHelper := secrets.NewSecretHelper(c.secretProvider, runNamespace, targetClient)
+
+	pipelineCloneSecretName, err := c.copyPipelineCloneSecret(pipelineRun, secretHelper)
+	if err != nil {
+		return errors.Wrap(err, "failed to copy pipeline clone secret")
+	}
```

Comment:
In case the secret does not exist, the PipelineRun's status should be set (like for missing pipeline secrets below).

Reply:
This is done in the copySecrets function

Reply:
True. But there's another case: if `pipelineRun.GetPipelineRepoServerURL()` called by `copyPipelineCloneSecrets()` fails,the PipelineRun's status is not updated.